### ISSUE TITLE
Add confirmation step with complete summary of what will be sent

### ISSUE THREAD

**Issue by stvnrlly** [GSA/notifications-admin#815]
When sending, it's nice to know exactly what will happen. Maybe something like "This will send to [x] phone numbers, starting at [time]." accompanied by "Start sending" button?

This would be a new step in the sending flow, following the selection of a sending time.

To complete:
- [ ] Add a confirmation step, with a new view, following the time selection
- [ ] Switch time selection to be something like "Set time" rather than "Send"

**Comment by tdlowden** [comment]
One thing we definitely need here is a calculation of message parts.

>This batch will be sent to X phone numbers and account for X message parts. Would you like to proceed?

**Comment by em-herrick** [comment]
@jonathanbobel @heyitsmebev @maomeara63 confirmation page user story has been built out on this ticket. Looking forward to hearing your thoughts on process/tasking that I built out here + any of the content in the ticket specifically!

**Comment by heyitsmebev** [comment]
After our meeting 1/32/2024, here are some notes we gathered and validated.

[Design Notes](https://docs.google.com/document/d/1COPxrjr9FNdqUocV0oUM65oWE405sBbN0Zf_lHn6-Fo/edit?usp=drive_link)

[Figma Design](https://www.figma.com/file/aAWhlPzOP6kfVFnCCAfSZZ/notify.gov?type=design&node-id=375-808&mode=design&t=VaCnAvxVJ7lXdNbD-0)

**Comment by heyitsmebev** [comment]
Based on UX meeting 2/7 

Revision notes: 

- [ ] Remove the "Send list Snow Day Alert Testing.csv" section.
- [ ] Change "Time" to "Scheduled" and display as “Now” or specify a later time.
- [ ] Update header from “Preview List” to “Recipients List” and add disclaimer: "Note: Only the first 5 rows are displayed here."
- [ ] Remove “to” within detail information for batch jobs  
- [ ] In the "Parts Messaging" section, make the counts for recipients, message parts, and message parts remaining bold.
400 Recipients (make bold)
800 message parts (make bold) 
“249,200 message parts remaining” (make bold) 

- [ ] Based on users actions, display either "Send" for now or "Schedule" for later as button text on the preview page. 
- [ ] Remove “template” from the header 
- [ ] Add “template” title in the details information for all preview pages

**Comment by heyitsmebev** [comment]
Based on UX meeting 2/14

**Confirmation page:** 
- [ ] change the date to "America" February 14 rather than 14 February
- [ ] Update the alert to say: 
H4: Your text has been scheduled
Body: [Template name] from texting service was scheduled [date] [time]
- [ ] Update page title to "Message status"
- [ ] Add in Message preview, send list, and SMS status 
- [ ] Remove the status section of the page after a certain amount of time
![Image](https://github.com/GSA/notifications-admin/assets/53159604/48c2d82c-d936-4d3a-adb6-89d6a8dd3814)



**Preview page:** 
- [ ] Update the button when a user "Schedules" a future send time to "schedule" and not "scheduled" 
- [ ] Adjust padding within the "message information details" at the top of the page

**Comment by maomeara63** [comment]
@heyitsmebev I'm not sure the "schedules" line is exactly right. Can you let me know when you have time and we can look at it together?

**Comment by maomeara63** [comment]
@heyitsmebev That's exactly right. Thank you!

**Comment by heyitsmebev** [comment]
Updated the preview page with screenshots showing the new changes related to padding/spacing. Additionally, the button text has been changed from 'scheduled' to 'schedule'.

Below is a screenshot of the page without "placeholder parts text". Once that ticket is complete, I'll add the parts messaging. Also the date/time in "scheduled" will be updated in another ticket to a more "human" readable format. 

![Image](https://github.com/GSA/notifications-admin/assets/53159604/1c4f18e7-3f4e-4bbd-b992-627ad15b3b8a)

**Comment by maomeara63** [comment]
The parts messaging in the screenshots, above, looks good.

**Comment by em-herrick** [comment]
Spacing looks a lot better!

**Comment by heyitsmebev** [comment]
> **Confirmation page:**
> 
> * [x]  change the date to "America" February 14 rather than 14 February
> * [x]  Update the alert to say:
>   H4: Your text has been scheduled
>   Body: [Template name] from texting service was scheduled [date] [time]
> * [x]  Update page title to "Message status"
> * [x]  Add in Message preview, send list, and SMS status
> * [x]  Remove the status section of the page after a certain amount of time

**_I've made changes to the confirmation/message status page, and here's how it currently looks. A couple of things to note:_** 

---
## These screenshot are from one-off send flow when a user schedules a message. 
Even though a user in this step would not upload a CSV file, sometimes the backend will create a new CSV file for one-off messages. There's a condition here that will display a Recipient list section _**if**_ there's a CSV file.


> :memo: **Note:** - This is an example of the Recipient list section
![Image](https://github.com/GSA/notifications-admin/assets/53159604/94b97673-cd3b-49c2-b00c-fadc0461a2e3)


:point_down: **Screenshot:** - Below is a screenshot of the one-off message without the Recipients list section
![Image](https://github.com/GSA/notifications-admin/assets/53159604/1b8c8e0d-07f3-4d61-a258-5088fa31e76f)

:point_down: **Screenshot:** - Below is a screenshot of the one-off message WITH the Recipients list section
![Image](https://github.com/GSA/notifications-admin/assets/53159604/8cb53577-2a2b-4b0c-8aac-45950c8c63c5)

---
## These screenshot are from upload csv file send flow when a user sends a message. 

:point_down: **Screenshot:** - This screenshot demonstrates what the message status page will look like when a user sends a message "now"
![Image](https://github.com/GSA/notifications-admin/assets/53159604/56df9a57-ecb7-4d75-aa87-588718f13831)


:point_down: **Screenshot:** - This screenshot demonstrates what the message status page will look like when a user returns to this page NOT from the send-flow, i.e., from the recent batch table:
![Image](https://github.com/GSA/notifications-admin/assets/53159604/64ceb412-87dd-4bcc-9e7e-b7d8218e05d8)

**Comment by jonathanbobel** [comment]
@heyitsmebev very small css update - less padding or margin between the header "Recipients List" and the file name and icon. Would make it consistent with the vertical spacing between "Message" and the message bubble.

**Comment by maomeara63** [comment]
@heyitsmebev This is looking great. A couple of things:

- Make Recipients list Recipient list (singular). It isn't inaccurate plural, but it works for both one and more than one messages as singular.
- Change the heading SMS status to Message status. It seems to be changed in one spot, but not another?
- And then the dates and times...

Thanks!

**Comment by heyitsmebev** [comment]
Thanks for your valuable feedback @maomeara63 

- We have "Message status" up on top of the page in the header. If we change SMS status to Message status, we would be repeating ourselves. I'm not sure what you mean by it changed in "one spot, but not another"

![Image](https://github.com/GSA/notifications-admin/assets/53159604/d38c335b-ca86-468a-93e0-ca021e29da38)

- Which dates and times are you referring to? The top banner or the dates and times in the status table (if so, that's out of the scope of this ticket).

**Comment by maomeara63** [comment]
Good point about the headings, @heyitsmebev. Leave SMS Status.

Is the date at the bottom just above Cancel sending part of another ticket?

**Comment by em-herrick** [comment]
> Good point about the headings, @heyitsmebev. Leave SMS Status.
> 
> Is the date at the bottom just above Cancel sending part of another ticket?

Perhaps we change to `Delivery Status`

**Comment by em-herrick** [comment]
@heyitsmebev happy to QA this as it's deployed in production! LMK

**Comment by ccostino** [comment]
There are some specific elements to this that are still outstanding; however, we completed this work and have a much improved confirmation page and are now in a new phase with the product, so closing this issue out as completed!
